compaction/compaction_manager.cc

```diff
@@ -1480,7 +1480,7 @@ const std::vector<sstables::compaction_info> compaction_manager::get_compactions
     };
     using ret = std::vector<sstables::compaction_info>;
     return boost::copy_range<ret>(_tasks | boost::adaptors::filtered([t] (const shared_ptr<task>& task) {
-                return (!t || task->compacting_table() == t) && task->compaction_running();
+                return (!t || &task->compacting_table()->as_table_state() == t) && task->compaction_running();
```

Comment:
No, that's really bad.
The task should work on a table_state (actually on a compaction group :))
and should expose it via a `task::compaction_group()` method that can be used here.

Note: with that I'm not sure we'll need `compacting_table()` anymore.
the compaction_group object should have a back reference to the table as compaction groups represent a subset of the table's data.

Reply:
I think it will be cleared by following patches that change task::compacting_table() to return a table_state.

Reply:
@avikiviy is right. I had to do this for making the series easier to read. I was only able to convert task much later due to cyclic dependencies.